src/planscape/planning/views.py

```diff
@@ -939,3 +942,115 @@ def create_shared_link(request: Request) -> Response:
     except Exception as e:
         logger.error("Error creating shared link: %s", e)
         raise
+
+
+class PlanningAreaNotes(APIView):
+    def post(self, request: Request, planningarea_pk: int):
+        user = request.user
+        if not user.is_authenticated:
+            return Response(
+                {"error": "Authentication Required"},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
+        note_data = request.data.copy()
+        note_data["planning_area"] = planningarea_pk
+
+        try:
+            serializer = PlanningAreaNoteSerializer(
+                data=note_data, context={"user": user}
+            )
+            serializer.is_valid(raise_exception=True)
+            if not PlanningAreaNotePermission.can_add(user, serializer.validated_data):
+                return Response(
+                    {"error": "Authentication Required"},
+                    status=status.HTTP_403_FORBIDDEN,
+                )
+            new_note = serializer.save()
+            out_serializer = PlanningAreaNoteSerializer(new_note)
+            return Response(
+                out_serializer.data,
+                content_type="application/json",
+                status=status.HTTP_201_CREATED,
+            )
+        except Exception as e:
+            logger.error("Error creating PlanningAreaNote: %s", e)
+            raise
+
+    def get(
+        self, request: Request, planningarea_pk: int, planningareanote_pk: int = None
+    ):
+        user = request.user
+        if not user.is_authenticated:
+            return Response(
+                {"error": "Authentication Required"},
+                status=status.HTTP_401_UNAUTHORIZED,
+            )
+
+        try:
+            if planningareanote_pk:
+                planningareanote = PlanningAreaNote.objects.get(
+                    id=planningareanote_pk, planning_area=planningarea_pk
+                )
+                if not PlanningAreaNotePermission.can_view(user, planningareanote):
+                    return Response(status=status.HTTP_403_FORBIDDEN)
+                serializer = PlanningAreaNoteSerializer(
+                    instance=planningareanote, many=False
+                )
+                return Response(serializer.data)
+
+            else:
+                planningarea = PlanningArea.objects.get(id=planningarea_pk)
+                # TODO: all of the notes in this query will have the same PlanningArea, so for now,
+                #  having view access to the PlanningArea means a viewer could see the notes,
+                #  so we could just check for access to Planningarea here.
+                # However, if this permission structure *ever* changes we'll need to update this, and that may not be easy to remember
+                # One alternative here is to query for can_view on each note, but it's a lot of queries.
+                # Another alternative? write a function to intelligently check for permissions for sevearl notes?
```

Comment:
We're following this permissions patten [ in scenarios](https://github.com/OurPlanscape/Planscape/blob/62fead164b8d9de37b1f7a57d700f5386a1d0652/src/planscape/planning/views.py#L812) now, so if we're fine with it there, maybe this isn't a huge issue.